src/bindings/xfconf/elektra-xfconf-channel.c

```diff
@@ -0,0 +1,1121 @@
+	}
+	g_debug ("Refreshing keyset for channel %s from kdb, currently %ld keys", channel_name, ksGetSize (channelPair->keySet));
+	int getStatusCode = kdbGet (elektraKdb, channelPair->keySet, parentKey);
+	g_debug ("Refreshed keyset (status code was %d) now has %ld keys", getStatusCode, ksGetSize (channelPair->keySet));
+	if (already_locked != 1)
+	{
+		release_channel_lock ()
+	}
+
+	switch (getStatusCode)
+	{
+	case -1:
+		g_warning ("There was a failure fetching the keys");
+		break;
+	case 0:
+		g_debug ("The keyset did not change");
+		break;
+	case 1:
+		g_debug ("Retrieved the keyset");
+		break;
+	case 2:
+		g_debug ("No data was loaded from storage");
+		break;
+	default:
+		g_warning ("An unknown status code(%d) occurred during keyset fetch", getStatusCode);
+	}
+	if (XFCONF_DEBUG_LOG_FOUND_KEYS == 1)
+	{
+		Key * cur;
+		if (already_locked != 1)
+		{
+			require_channel_read_lock ()
+		}
+		ssize_t keySetSize = ksGetSize (channelPair->keySet);
+		g_debug ("KeySet has %ld keys", keySetSize);
+		for (elektraCursor i = 0; i < keySetSize; i++)
+		{
+			cur = ksAtCursor (channelPair->keySet, i);
+			g_debug ("Found key: %s", keyName (cur));
+		}
+		if (already_locked != 1)
+		{
+			release_channel_lock ()
+		}
+		free (kdbKeyName);
+	}
+	if (reference != 0)
+	{
+		return channelPair->keySet;
+	}
+	if (already_locked != 1)
+	{
+		require_channel_read_lock ()
+	}
+	KeySet * deepCopy = ksDeepDup (channelPair->keySet);
+	if (already_locked != 1)
+	{
+		release_channel_lock ()
+	}
+	return deepCopy;
+}
+
+/**
+ * Convert a string to a GValue - This function tries to parse the content of `new_value` and stores it in the `g_value` pointer.
+ * If that is not possible, e.g. the `g_value` has an unknown type, this function does nothing.
+ * The desired type is taken from the `g_value`.
+ * @param g_value the value to parse the string to - not null
+ * @param new_value the string of the new value - not null
+ * @return 0 on success and 1 otherwise
+ */
+static int g_value_convert_string (GValue * g_value, const char * new_value)
+{
+	trace ();
+	switch (g_value->g_type)
+	{
+	case G_TYPE_INT:
+		g_value_set_int (g_value, (int) strtol (new_value, NULL, 10));
+		break;
+	case G_TYPE_INT64:
+		g_value_set_int64 (g_value, strtol (new_value, NULL, 10));
+		break;
+	case G_TYPE_UINT:
+		g_value_set_uint (g_value, (uint) strtoull (new_value, NULL, 10));
+		break;
+	case G_TYPE_UINT64:
+		g_value_set_uint64 (g_value, strtoull (new_value, NULL, 10));
+		break;
+	case G_TYPE_LONG:
+		g_value_set_long (g_value, strtoll (new_value, NULL, 10));
+		break;
+	case G_TYPE_ULONG:
+		g_value_set_ulong (g_value, strtoull (new_value, NULL, 10));
+		break;
+	case G_TYPE_FLOAT:
+		g_value_set_float (g_value, strtof (new_value, NULL));
+		break;
+	case G_TYPE_DOUBLE:
+		g_value_set_double (g_value, strtod (new_value, NULL));
+		break;
+	case G_TYPE_BOOLEAN:
+		if (strcmp (new_value, "TRUE") == 0 || strcmp (new_value, "true") == 0 || strcmp (new_value, "True") == 0)
```

Comment:
elektraStrCaseCmp? Or is this really the definition of bool in xfconf?

Reply:
Done with 35b3c5a02291276ff9108da08cd0b2d9525b101a.